sklearn/utils/estimator_checks.py

```diff
@@ -356,6 +356,23 @@ def _generate_class_checks(Estimator):
     yield from _generate_instance_checks(name, estimator)
 
 
+def _mark_xfail_checks(estimator, check, pytest):
+    """Mark estimator check pairs with xfail"""
+
+    xfail_checks = _safe_tags(estimator, '_xfail_test')
+    if not xfail_checks:
+        return estimator, check
+
+    try:
+        check_name = _set_check_estimator_ids(check)
+        msg = xfail_checks[check_name]
+        return pytest.param(
+            estimator, check, marks=pytest.mark.xfail(reason=msg))
+
+    except KeyError:
```

Comment:
I'm not very keen on this. KeyError can happen for a variety of reasons say in `pytest.params` in which case it will silently pass that error.

I would rather we did, 
```
msg = xfail_checks.get(check_name, None)
if msg is not None:
```
or something similar.

Generally the case of a KeyError is the default case here (since most tests are not xfailed) and triggering/catching an exception being expensive is suboptimal https://en.wikipedia.org/wiki/Python_syntax_and_semantics#Exceptions

Reply:
Makes sense in this case. I have made the change.